File: backend/src/main/java/com/talenthub/service/ProfileService.java
package com.talenthub.service;

import com.talenthub.dto.ArtistProfileDto;
import com.talenthub.dto.OrganizerProfileDto;

public interface ProfileService {
    ArtistProfileDto getArtistProfile(Long userId);
    ArtistProfileDto updateArtistProfile(Long userId, ArtistProfileDto dto);
    OrganizerProfileDto getOrganizerProfile(Long userId);
    OrganizerProfileDto updateOrganizerProfile(Long userId, OrganizerProfileDto dto);
}

File: frontend/src/app/core/services/profile.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ProfileService {
  private apiUrl = `${environment.apiUrl}/profiles`;

  constructor(private http: HttpClient) {}

  getArtistProfile(): Observable<any> {
    return this.http.get(`${this.apiUrl}/artist`);
  }

  updateArtistProfile(profile: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/artist`, profile);
  }

  getOrganizerProfile(): Observable<any> {
    return this.http.get(`${this.apiUrl}/organizer`);
  }

  updateOrganizerProfile(profile: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/organizer`, profile);
  }
}
